app/controllers/bikes_controller.rb

```diff
@@ -153,51 +153,107 @@ def create
+    @edit_templates = edit_templates
+
+    is_valid, @edit_template = target_edit_template(params[:page], edit_templates.keys)
+
+    if !is_valid
+      redirect_to edit_bike_url(@bike, page: @edit_template) and return
+    end
+
     if @edit_template == "photos"
-      @private_images = PublicImage.unscoped.where(imageable_type: "Bike").where(imageable_id: @bike.id).where(is_private: true)
+      @private_images =
+        PublicImage
+          .unscoped
+          .where(imageable_type: "Bike")
+          .where(imageable_id: @bike.id)
+          .where(is_private: true)
     end
+
     render "edit_#{@edit_template}".to_sym
   end
 
   def update
     begin
-      @bike = BikeUpdator.new(user: current_user, bike: @bike, b_params: permitted_bike_params.as_json, current_ownership: @current_ownership).update_available_attributes
+      @bike = BikeUpdator.new(
+        user: current_user,
+        bike: @bike,
+        b_params: permitted_bike_params.as_json,
+        current_ownership: @current_ownership,
+      ).update_available_attributes
     rescue => e
       flash[:error] = e.message
     end
+
     @bike = @bike.decorate
+
     if @bike.errors.any? || flash[:error].present?
       edit and return
     else
       flash[:success] = "Bike successfully updated!"
       return if return_to_if_present
-      redirect_to edit_bike_url(@bike, page: params[:edit_template]) and return
-    end
-  end
 
-  def edit_templates_hash
-    stolen_type = @bike.recovered ? "Recovery" : "Theft"
-    hash = {
-      root: "Bike Details",
-      photos: "Photos",
-      drivetrain: "Wheels + Drivetrain",
-      accessories: "Accessories + Components",
-      ownership: "Transfer Ownership",
-      groups: "Groups + Organizations",
-      remove: "Hide or Delete",
-      stolen: (@bike.stolen ? "#{stolen_type} details" : "Report Stolen or Missing"),
-    }
-    # To make stolen the first key if bike is stolen. using as_json for string keys instead of sym
-    (@bike.stolen ? hash.to_a.rotate(-1).to_h : hash).as_json
+      # When reporting a bike as stolen, redirect to the recovery form.
+      # The recovery form redirects to the stolen form and is handled by
+      # CoffeeScript in pages/bikes/edit_stolen.coffee
+      target_template = (params[:edit_template] == "report_stolen") ?
+        "report_recovered" : params[:edit_template]
+      redirect_to edit_bike_url(@bike, page: target_template) and return
+    end
   end
 
   def edit_templates
-    @edit_templates ||= edit_templates_hash
+    return @edit_templates if defined?(@edit_templates)
+    @theft_templates = theft_templates
+    @bike_templates = bike_templates
+    @edit_templates = @theft_templates.merge(@bike_templates)
   end
 
   protected
 
+  # Determine the appropriate edit template to use.
+  def target_edit_template(requested_page, valid_pages)
+    default_page = (@bike.stolen? ? :theft_details : :bike_details)
+    return [true, default_page] if requested_page.blank?
```

Comment:
I would prefer not to use parallel assignment. Can you make this return either a hash or a single value?

Reply:
Updated in 94e14ef